### PR TITLE
Rdm 2314 CasePaymentHistoryViewer BE Type

### PR DESCRIPTION
**Pull request by andrewfolga**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-2314

### Change description ###
Rdm 2314 CasePaymentHistoryViewer BE Type

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/validation/eventcasefield/EventCaseFieldCasePaymentHistoryViewerCaseFieldValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+
+public class EventCaseFieldCasePaymentHistoryViewerCaseFieldValidatorTest {
+
+    public static final String CASE_PAYMENT_HISTORY_VIEWER = "CasePaymentHistoryViewer";
+    private EventCaseFieldCasePaymentHistoryViewerCaseFieldValidator classUnderTest =
+        new EventCaseFieldCasePaymentHistoryViewerCaseFieldValidator();
+
+    @Test
+    public void shouldPassValidationIfFieldIsNotLabelType() {
```
Comment (andrewfolga) [inline domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/validation/eventcasefield/EventCaseFieldCasePaymentHistoryViewerCaseFieldValidatorTest.java]: change test name

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/eventcasefield/EventCaseFieldCasePaymentHistoryViewerCaseFieldValidator.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package uk.gov.hmcts.ccd.definition.store.domain.validation.eventcasefield;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationResult;
+import uk.gov.hmcts.ccd.definition.store.repository.DisplayContext;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.EventCaseFieldEntity;
+
+@Component
+public class EventCaseFieldCasePaymentHistoryViewerCaseFieldValidator implements EventCaseFieldEntityValidator {
```
Comment (mario-paniccia) [inline domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/eventcasefield/EventCaseFieldCasePaymentHistoryViewerCaseFieldValidator.java]: can't this validator use the same approach used by EventCaseFieldDisplayContextValidatorImpl of defining the exception as an inner static class? 
This way we don't have to have the exception as a new type

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/eventcasefield/EventCaseFieldCasePaymentHistoryViewerCaseFieldValidator.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    @Override
+    public ValidationResult validate(EventCaseFieldEntity eventCaseFieldEntity,
+                                     EventCaseFieldEntityValidationContext eventCaseFieldEntityValidationContext) {
+        ValidationResult validationResult = new ValidationResult();
+
+        if ("CasePaymentHistoryViewer".equals(eventCaseFieldEntity.getCaseField().getFieldType().getReference())
```
Comment (mario-paniccia) [inline domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/eventcasefield/EventCaseFieldCasePaymentHistoryViewerCaseFieldValidator.java]: any way we can avoid hardcoding the filed type id here? 
Perhaps accessing it via FieldTypeUtils.BASE_CASE_PAYMENT_HISTORY_VIEWER or similar?

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/eventcasefield/EventCaseFieldCasePaymentHistoryViewerCaseFieldValidator.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        }
+
+        return validationResult;
+    }
+
+    private boolean isReadOnlyDisplayContext(EventCaseFieldEntity eventCaseFieldEntity) {
+        return eventCaseFieldEntity.getDisplayContext().equals(DisplayContext.READONLY);
+    }
+
+    private boolean isEmptyDisplayContext(EventCaseFieldEntity eventCaseFieldEntity) {
```
Comment (mario-paniccia) [inline domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/eventcasefield/EventCaseFieldCasePaymentHistoryViewerCaseFieldValidator.java]: isEmptyDisplayContext and isCasePaymentHistoryViewer are duplicated in the 3 validators. Could be pushed up to the common EventCaseFieldEntityValidator interface as defaults methods

**Pull request merged**